File: python/hashes/1679-max-number-of-k-sum-pairs.py
from typing import List


def maxOperations(self, nums: List[int], k: int) -> int:
    max_op = 0
    operations = defaultdict(int)
    
    for num in nums:
        if operations[num] > 0:
            operations[num] -= 1
            max_op += 1
        else:
            operations[k - num] += 1
    
    return max_op

File: cpp/hashes/1679-max-number-of-k-sum-pairs.cpp
#include <unordered_map>
#include <vector>

using namespace std;


int maxOperations(vector<int>& nums, int k) {
    int max_op = 0;
    unordered_map<int, int> mp;
    
    for (auto n : nums) {
        if (mp[n] > 0) {
            mp[n]--;
            max_op++;
        } else {
            mp[k - n]++;
        }
    }
    
    return max_op;
}